utils/S3UtilsMongoClient.js

```diff
@@ -74,6 +74,92 @@ class S3UtilsMongoClient extends MongoClientInterface {
         }
     }
 
+    async updateInflightDeltas(allMetrics, log) {
+        let cursor;
+        try {
+            const bucketKeys = [];
+
+            if (!allMetrics || !Array.isArray(allMetrics) || allMetrics.length === 0) {
+                return allMetrics;
+            }
+
+            const usersBucketCreationDatesMap = await this._getUsersBucketCreationDates(log);
+            Object.keys(usersBucketCreationDatesMap).forEach(bucketEntry => {
+                const bucketName = bucketEntry.split(constants.splitter)[1];
+                const bucketKey = `bucket_${bucketName}_${new Date(usersBucketCreationDatesMap[bucketEntry]).getTime()}`;
+                bucketKeys.push(bucketKey);
+            });
+
+            cursor = await this.getCollection(INFOSTORE).find({
+                _id: { $in: bucketKeys },
+            }, {
+                projection: {
+                    'usedCapacity._inflight': 1,
+                },
+            });
+
+            const inflights = await cursor.toArray();
```

Comment:
why do we convert to array: we should be able to directly call `custer.forEach(...)` ?